connector_salesforce/salesforce_account/importer.py

```diff
@@ -0,0 +1,191 @@
+        If a shipping address already exists we update it.
+        If no shipping data are present and a shipping adress exists
+        it will be unactivated
+
+        """
+        if not binding:
+            raise MappingError(
+                'No binding when mapping shipping address'
+            )
+        binding.ensure_one()
+        current_partner = binding
+        shipp_id = False
+        shipp_fields = (field for field in record
+                        if field.startswith('Shipping'))
+        if any(record[field] for field in shipp_fields):
+            if current_partner.sf_shipping_partner_id:
+                shipp_id = current_partner.sf_shipping_partner_id.id
+                self.session.env['res.partner'].write(
+                    [shipp_id],
+                    self._prepare_shipp_addresse_data(record, current_partner)
+                )
+            else:
+                shipp_id = self.session.env['res.partner'].create(
+                    self._prepare_shipp_addresse_data(record, current_partner)
+                ).id
+        else:
+            shipp_id = False
+            if current_partner.sf_shipping_partner_id:
+                self.session.env['res.partner'].write(
+                    [current_partner.sf_shipping_partner_id.id],
+                    {'active': False}
+                )
+        return {'sf_shipping_partner_id': shipp_id}
+
+    @only_create
+    @mapping
+    def backend_id(self, record):
+        return {'backend_id': self.backend_record.id}
+
+    @only_create
+    @mapping
+    def is_company(self, record):
+        return {'is_company': True}
+
+    @mapping
+    def country_id(self, record):
+        country_id = self._country_id(record, 'BillingCountryCode')
+        return {'country_id': country_id}
+
+    @mapping
+    def state_id(self, record):
+        state_id = self._state_id(record, 'BillingState', 'BillingCountryCode')
+        return {'state_id': state_id}
+
+    @mapping
+    def customer(self, record):
+        return {'customer': True}
+
+    @mapping
+    def active(self, record):
+        return {'active': True}
+
+    @mapping
+    def property_product_pricelist(self, record):
+        """Map property pricelist using current backend setting"""
+        currency_id = self.get_currency_id(record)
+        mapping = {rec.currency_id.id: rec.pricelist_version_id.id
+                   for rec in self.backend_record.sf_entry_mapping_ids}
+        price_list_version_id = mapping.get(currency_id)
+        if not price_list_version_id:
+            raise MappingError(
+                'No pricelist version configuration done for '
+                'currency %s and backend %s' % (
+                    record.get('CurrencyIsoCode'),
+                    self.backend_record.name
+                )
+            )
+        pl_model = 'product.pricelist.version'
+        price_list_version_record = self.session.env[pl_model].browse(
+            price_list_version_id
```

Comment:
Is this browse necessary? can't we get already the record from `sf_entry_mapping_ids` ?